packages/app/src/app/hooks/useCreateCheckout.ts

```diff
@@ -0,0 +1,43 @@
+import { useState } from 'react';
+import { useEffects } from 'app/overmind';
+import { dashboard } from '@codesandbox/common/lib/utils/url-generator';
+
+type CheckoutStatus =
+  | { status: 'idle' }
+  | { status: 'loading' }
+  | { status: 'error'; error: string };
+
+export const useCreateCheckout = (): [
```

Comment:
I see this is a single time use async request basically. But I'm still wondering how we can integrate it with the rest of the overmind state, or if it makes sense to integrated it. More like food for thought. We could technically have a separate namespace for the signup flow, where we keep track of all the client state as the user goes through the flow. But I also see the simplicity with this hook and it is somewhat similar with what we are used to from v2.

Reply:
@alexnm yep, I tried to follow the pattern for hooks that we have on v2 since we will eventually move there.